code/lib/core-events/src/errors/server-errors.test.ts

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable local-rules/no-uncategorized-errors */
+import { WebpackCompilationError } from './server-errors';
+
+describe('WebpackCompilationError', () => {
+  it('should correctly handle error with error property', () => {
+    const error = new Error('Custom error message');
+    const data = {
+      error,
+    };
+
+    const webpackError = new WebpackCompilationError(data);
+
+    expect(webpackError.message).toBe(error.message);
+  });
+
+  it('should correctly handle error with error within error', () => {
+    const error = new Error('Custom error message');
+    const data = {
+      error: new Error(),
+    };
+    data.error = error;
```

Comment:
Is this supposed to be `data.error.error = error`?

Reply:
great catch! fixed